src/core/reactor.cc

```diff
@@ -2218,6 +2219,45 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<group_details> reactor::grp_detail(std::string_view groupname) noexcept {
+    return futurize_invoke([groupname, this] {
+        return _thread_pool->submit<syscall_result_extra<struct group>>([groupname = sstring(groupname)] {
+            struct group *gg = ::getgrnam(groupname.c_str());
+            if (!gg) {
+                return wrap_syscall(-1, *gg); // Simulating failure
+            }
+            return wrap_syscall(0, *gg);
+        }).then([groupname = sstring(groupname)] (syscall_result_extra<struct group> gr) {
+            gr.throw_fs_exception_if_error("group failed", groupname);
+            struct group& g = gr.extra;
+            group_details g_details;
+            g_details.group_name = g.gr_name;
+            g_details.group_password = g.gr_passwd;
+            g_details.group_gid = g.gr_gid;
+            g_details.group_member = g.gr_mem;
+            return make_ready_future<group_details>(std::move(g_details));
+        });
+    });
+}
+
+future<> reactor::change_ownership_of_file(std::string_view filepath, uint64_t groupid) noexcept {
+    return futurize_invoke([filepath, groupid, this] {
+        return _thread_pool->submit<syscall_result<int>>([filepath = sstring(filepath), groupid] {
+            int chown_result = ::chown(filepath.c_str(), ::geteuid(), groupid);
+            return wrap_syscall(chown_result);
+        }).then([filepath = sstring(filepath), groupid] (syscall_result<int> sr) {
+            if (sr.result < 0) {
+                if (sr.error == EPERM) {
```

Comment:
IMO, we should leave it to the caller to throw the user friendly and length message. as the caller understand what is the best way to communicate with the user, seastar should focus on returning the accurate error. so,
```c++
throw_fs_exception_if_error("chown failed", filepath);
```
would suffice.